### PR TITLE
Fixed a crash on new map creation

### PR DESCRIPTION
**Pull request by Mailaender**
A regression from https://github.com/OpenRA/OpenRA/pull/10344.

### CODE REVIEW

File: OpenRA.Mods.Common/Widgets/Logic/Editor/SaveMapLogic.cs [hunk 1]
```diff
@@ -93,10 +93,16 @@ public SaveMapLogic(Widget widget, Action<string> onSave, Action onExit, Map map
...
+
+			if (map.Path != null)
+			{
+				var attr = File.GetAttributes(map.Path);
+				mapIsUnpacked = attr.HasFlag(FileAttributes.Directory);
+			}
+
 			var filename = widget.Get<TextFieldWidget>("FILENAME");
-			filename.Text = mapIsUnpacked ? Path.GetFileName(map.Path) : Path.GetFileNameWithoutExtension(map.Path);
+			filename.Text = mapIsUnpacked && map.Path != null ? Path.GetFileName(map.Path) : Path.GetFileNameWithoutExtension(map.Path);
```
Comment (obrakmann) [inline OpenRA.Mods.Common/Widgets/Logic/Editor/SaveMapLogic.cs]: This isn't needed. If `mapIsUnpacked` is true, `map.Path` cannot be null.

**Comment by obrakmann** [comment]
Other than that, :+1:

File: OpenRA.Mods.Common/Widgets/Logic/Editor/SaveMapLogic.cs (hunk 1 above)
Comment (Mailaender) [inline OpenRA.Mods.Common/Widgets/Logic/Editor/SaveMapLogic.cs]: I know, it is an extra safeguard for future changes.

Comment (obrakmann) [inline OpenRA.Mods.Common/Widgets/Logic/Editor/SaveMapLogic.cs]: It won't safeguard anything, `map.Path` is used on both sides of the `:`. Also both `Path.GetFilename` and `Path.GetFileNameWithoutExtension` work fine when you pass them a `null`. Please remove it.

Comment (Mailaender) [inline OpenRA.Mods.Common/Widgets/Logic/Editor/SaveMapLogic.cs]: Removed.

**Comment by RoosterDragon** [comment]
:+1:

**Comment by RoosterDragon** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/c77e365820c8408bf5874c37548ccb538e0457b5)

**Pull request merged**